### ISSUE TITLE
Gracefully handle invalid Location headers

### ISSUE THREAD

**Issue by mudge** [altmetric/embiggen#4]
While [RFC 7231 specifies that `Location` headers should be valid URI references](http://tools.ietf.org/html/rfc7231#section-7.1.2), we have seen headers with unescaped characters in them, e.g. http://shar.es/1p3DSI:

```shell
$ curl -I http://shar.es/1p3DSI                                                                                                                        HTTP/1.1 301 Moved Permanently
Location: http://paradrasi.gr/2015/04/19/ούτε-σεισμοί-ούτε-λιμοί-ούτε-καταποντ/
```

When passed to `URI`, this fails with an `InvalidURIError` as URIs must be ASCII-only.

**Comment by mudge** [comment]
A tricky issue here is that `URI.escape` and `URI.unescape` have actually been deprecated. See http://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape#answer-13059657 but the conclusion is to use [Addressable](https://github.com/sporkmonger/addressable) instead for encoding URIs.

**Comment by mudge** [comment]
Fixed in 0.2.0.
